Add mod count to mods page

Trial97 opened the pull request:
<!--
Hey there! Thanks for your contribution.

Please make sure that your commits are signed off first.
If you don't know how that works, check out our contribution guidelines: https://github.com/PrismLauncher/PrismLauncher/blob/develop/CONTRIBUTING.md#signing-your-work
If you already created your commits, you can run `git rebase --signoff develop` to retroactively sign-off all your commits and `git push --force` to override what you have pushed already.

Note that signing and signing-off are two different things!
-->
Fixes #451

Trial97 committed:
Added dynamic page extra info

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into mods_txt

getchoo commented on launcher/ui/pages/instance/ExternalResourcesPage.cpp:
```diff
@@ -248,6 +256,15 @@ bool ExternalResourcesPage::onSelectionChanged(const QModelIndex& current, const
+    if (ui && ui->treeView && ui->treeView->selectionModel()) {
+        auto selection = m_filterModel->mapSelectionToSource(ui->treeView->selectionModel()->selection()).indexes();
+        if (auto count = std::count_if(selection.cbegin(), selection.cend(), [](auto v) { return v.column() == 0; }); count != 0)
+            return tr("[%1 installed, %2 selected]").arg(m_model->size()).arg(count);
```
```suggestion
            return tr(" [%1 installed, %2 selected]").arg(m_model->size()).arg(count);
```
this looks a little cramped imo
![image](https://github.com/PrismLauncher/PrismLauncher/assets/48872998/14eec71f-d73c-4923-ac31-83097d98e9bc)

TheKodeToad replied:
I think rounded brackets would be better

getchoo reviewed (changes requested):
i'm not exactly sure what's going on here: in my testing the mods page (and only the mods page) will have the install/select count appear when you first view it, but if you switch to a different tab like resource packs and go back, it won't appear at all until you have selected a mod.

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into mods_txt

Trial97 committed:
Updated text and fixed mod page text update

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 commented:
I fixed the mentioned comment but in order to fix all of them I removed a line that was prefixed by the following comment: 
```
// Prevent a weird crash when trying to open the mods page twice in a session o.O
disconnect(mods.get(), &ModFolderModel::updateFinished, this, 0);
```

I could not get it to crash as the description is vague. But if somebody knows why that line was added let me know.

getchoo commented:
it seems @flowln was the one who left this (sorry for the ping, not sure who else to ask :p). could you shed some light on this so we can see if it's still needed?

flowln commented:
The description isn't vague, it is just as simple as it sounds: Without that line, it used to be the case that the application would crash if you opened the mods page, switched pages, and returned back to the mods page. I suppose the underlying cause of the crash got fixed at some point, so the line doesn't do much now. Nice!

Also, you can probably remove `mods` from the lambda now.

getchoo commented:
thank you so much! just wanted to make sure :)

getchoo reviewed (approved):
i just tested this again as well and it seems to good to go! the only other thing i would recommend is following flow's suggestion

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into mods_txt

Trial97 committed:
Removed mods from lambda

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

DioEgizio reviewed (approved):
seems to work as expected

getchoo commented:
thanks!
